OpenTrons/opentrons-api
# feat(protocol-designer, components): allow trash to be movable

closes RAUT-758 and RAUT-753 and RAUT-719

# Overview

This PR allows the trash to be movable for Flex protocols and also wires up a modal for editing the waste chute. Also, a user is required to select either a waste chute or a trash bin. NOTE: no trash/waste chute is not supported until post PD 7.1 since return tip is currently out of scope, you can only drop tip to a trash bin or waste chute.

# Test Plan

- Turn the deck configuration FF flag on.
- Make a Flex protocol and don't add the trash bin but add a waste chute. Create the protocol
- in the Additional Items section, see that the Trash bin is selectable. (I am stubbing in a gripper asset for now). Click add and see the slots available in the modal. Select a slot and click save.
- try editing and removing the trash slot
- mess around with adding/deleting the trash bin and waste chute, make sure the `remove` button disables with a tooltip appropriately. A user is required to add one or the other for not since return tip to tiprack will not be supported in 7.1.
- when you have the trash in a slot, go to the deck map and see that it is rendered in the correct slot that you selected
- look at the module edit modals and the waste chute modal, i changed the layout to match the trash modal. Make sure the behavior works as normal and you can edit/add/delete modules. Note that the TC GEN2 slot only highlights B1, I will go back later to fix that.
- now, go create an OT-2 protocol, see that the edit modules modal works as expected, check the TC modal and see that all 4 slots are highlighted.

# Changelog

- make `setSelectedLocation` prop optional in `DeckLocationSelect` and add an optional `theme` prop that defaults to `default` with the purple theme
- allow modules to be placed in slot A3 if they support it
- create Trash modal and update the edit modules modal to have the same design
- extend `AdditionalItemsRow` to support trash bin
- add a disabled reason in `ModuleAndOtherTile` with a tooltip requiring you to select a trash or waste chute
- update and add tests

# Review requests

see test plan

# Risk assessment

low
